### DISCUSSION TITLE
How do i load Vector or Raster geo map from OpenStreetMap?

### DISCUSSION THREAD

**Discussion by afidegnum** [fltk-rs/fltk-rs#1525] [Q&A]
HI, 
I just discovered this crate and I'm really enthused. Can you please guide on how to load maps with zoom and waypoints into fltk? 

what do I need to make it work?

**Comment by MoAlyousef** [comment]
Hello

Unfortunately I don’t have experience with dealing OpenStreetMap. A cursory look seems you can use their api to download an image, and use a special OSM file (json or xml) to extract waypoints and their corresponding latitude/longitude. 
My naive approach would be to display the image in a frame, try to map the extents of the frame to the latitude and longitude. Transform the actual waypoint coordinates to the x,y axis of the image to place some other icon image on top of the map image. 
For zooming I would just rescale the image by a certain factor. 

Maybe other users do have experience with it and can offer better insights.
